languages/csharp/exercises/concept/properties/Properties.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+
+public enum Units
+{
+    Kilograms
+    ,Pounds
+}
+
```

Comment:
I don't think that is great from a quality control point of view.  Covid-19, Code that doesn't compile, whatever next?

Reply:
I'm not sure I understand the quality control issue. What do you mean by that? My main issue is that pre-defining the properties gives away much too much IMHO. There is hardly any syntax left for the students to be engaged with. It feels like we "pre-cooked" the solution, and personally I've had terrible experiences with learning things that way.

p.s. the covid-19 reference feels inappropriate to me to be honest.

Reply:
I just commented on this below before I saw this discussion.

So I'm a little split on this. By giving the get/set syntax do we give away a bit too much maybe? My question would be are people still writing the properties, or just filling them in? If they're just filling them in, then will they be learning as much about properties? Will it stick as much? So my instinct says to remove those stubs.

However, I'd want to understand what the error that the first test will give will be if they're not defined and whether it's C#-beginner-parseable or not. I don't think that the code **has** to compile - but if it doesn't I'd want the errors to be sensible. For the Ruby ones, for example, I'm leaving methods undefined because the runtime error is "method not defined - xyz", which is pretty clear.

Reply:
The error message will look like this:

```
'WeighingMachine' does not contain a definition for 'InputWeight' and no accessible extension method 'InputWeight' accepting a first argument of type 'WeighingMachine' could be found (are you missing a using directive or an assembly reference?)
```

Reply:
I could remove the extension method from the error message bit in the test runner if need be.

Reply:
1. QA: The advantage of compiled projects is that we know the all 3 files are fully consistent.  I know it’s not fool proof and there execution risk in copying the code about but it gives me a good feeling.  What could go wrong?  Property/method signatures in the test file could be changed, class and enum identifiers  could be changed.  Other typos in the main source file.
2. Sentiment: Our target audience is comprised of professional programmers.  We vary in our outlook but I suspect I represent a decent number in saying that anyone who gives me a project that does not compile is not my friend.
3. Learning: the vast majority of students will start the exercise knowing that a property has a get and a set accessor.  What we are teaching them is the subtleties of access control and the potential flexibility of  properties.  The one giveaway is the signature of the reference property - USDisplayWeight but even there the main point is to introduce them to the idea of spinning up an object in the get accessor which is not given away.

Reply:
These are all valid points, even though I still feel that the non-compiling version is better in _teaching_ about properties. But let's get some other C# maintainers to chip in: @robkeim @wolf99 @EarthlingRich what do you think of the above discussion about having stub properties or not?

Reply:
I see merit in both sides of the discussion here. However I would lean towards taking the stubs out, provided this exercise comes early in a student's progression (as I expect it does) - a student may only be learning the ropes of what Exercism asks them to do and how it asks them so may not immediately see anything missing with the stubs present. As  it is not the first exercise in the track, I think we can discount the idea that a large set of students may think something is wrong with their build machinery when it fails to compile (I would consider it to be an extra pointer to the fact that the exercise is requiring some change from them). It is a fine line between discouraging a student and spoon feeding them when early in a track. I cannot escape the feeling that writing the mutator and accessor outlines in an exercise on properties is on the spoon-feeding side!

There are good clear examples in the introduction and the hints file has plenty of good resources, so the student is not left without information on how to get started in this respect.

Reply:
After reading through these comments and thinking about this, my vote is for removing the stubs and leaving code that doesn't compile. If the target audience were people learning programming, I think I would lean towards having the stubs in, but given we're targeting experienced programmers that are just learning a new language they should be all too familiar with either compilation or run time errors.

Part of learning a programming language is learning the scaffolding of that language and not having the properties defined provides a teaching moment for some of the scaffolding as well as compilation errors which is also a reality of working in a compiled language. I personally feel having the properties pre-defined goes too far in the spoon-feeding direction.